File: Data Structures & Algorithms/counting-bits/submission-1.cpp
class Solution {
public:
    vector<int> countBits(int n) {
        int offset = 1;
        vector<int> result;
        result.push_back(0);
        for(int i = 1; i <= n; i++) {
            if(i == offset * 2) {
                offset *= 2;
            }
            int k = 1 + result[i - offset];
            result.push_back(k);
        }
        return result;
    }
};

File: Data Structures & Algorithms/task-scheduling/submission-1.py
from collections import defaultdict
import heapq
from collections import deque

class Solution:
    def leastInterval(self, tasks: List[str], n: int) -> int:
        frequencies = defaultdict(int)
        for task in tasks:
            frequencies[task] += 1
        pq = []
        #max heap = min heap but multiply values by -1

        for key, val in frequencies.items():
            heapq.heappush(pq, [val * -1, key])
        k = 0
        result = 0
        queue = deque()
        while pq or queue:
            if k == n + 1:
                k = 0
                while queue:
                    top = queue.popleft()
                    heapq.heappush(pq, top)
            if pq:
                top = heapq.heappop(pq)
                top[0] += 1
                if top[0] != 0:
                    queue.append(top)
            result += 1
            k += 1
        return result

